### DISCUSSION TITLE
Rotary firing mode selection

### DISCUSSION THREAD

**Discussion by nomakewan** [gpstar81/GPStar-proton-pack#303] [Ideas]
As inspired by Kaden's TVG build, I have been wanting to implement a rotary selection for firing mode. This is actually incredibly slick because it solves a lot of problems all at once. Essentially if the top switch is down (same way quick vent works) then the top knob on the wand would scroll through all the firing modes, while the barrel wing button goes straight to settings and back. When the switch is up, the barrel wing button is free to be used for other purposes. Here are the benefits:

-Rotary selection imitates the mouse scroll-wheel selection that was in the 2009 video game.
-Dedicating the rotary wheel to firing mode selection means accessing the settings menu can now be done with a single press instead of having to cycle through every single firing mode first.
-Keeping this behavior tied to the top switch means we still have power level selection available, and also means that the barrel wing button is freed to be used as an alternate firing switch (incredibly useful for adding the remaining missing TVG alt fire options).

Sounds great! But as you may have noticed, this functionality works very much like quick vent does. And quick vent is not available the "Prop Original" function mode because its switch sequence literally does not allow for such a thing. And just like quick vent, this solution does not work for "Prop Original" as-is. After spending a while thinking about it I came up with 3 potential solutions, but none of them are perfect. So to get feedback, I'll put them here and see what everyone thinks.

Solution 1: Put it in anyway in MODE_ORIGINAL even though the wand will be MODE_OFF, so that if you flip Activate off but the other two switches are on, the rotary knob switches firing modes and the alt switch goes straight to settings. The obvious disadvantage to this solution is that to change firing modes you would now have to completely shut down the pack (since that's what flipping Activate off does) where previously you switch firing modes with the alt fire button when everything is on in both MODE_ORIGINAL and MODE_SUPER_HERO.

Solution 2: Have it be only applicable for Super Hero mode, like quick vent already is. This certainly works, but it also means that there would be two completely different control schemes people would have to learn, and on top of that it bloats the code quite a bit (since both the old code and new code would have to coexist depending on which operation mode the wand is in). On top of that it means MODE_ORIGINAL wouldn't be able to use the alternate firing modes (boson dart, slime tether, etc). I very much do not like this solution.

Solution 3: Remove video game modes from MODE_ORIGINAL entirely. This swiftly solves 1 and 2 and can be very easily implemented (the checks for this are literally already in the code now), but there may be some people who for whatever reason wanted to run video game functionality alongside the original prop functionality. Out of the 3 I personally like this one the best (since if you're using MODE_ORIGINAL shouldn't you be using CTS or CTS Mix to imitate the films anyway?), but I can see where it would be contentious.

Let me know your thoughts!

**Comment by gpstar81** [comment]
I like Solution #3 the most. Mode Original in all intents and purposes is directly based on original documentation of the functionality of the props. 


Perhaps run a poll in the group to see what others prefer?

**Comment by nomakewan** [comment]
For those who might not be a part of the GPStar FB Group, we did host a poll and it was essentially 20/80 Solution 1/Solution 3. So with the majority of users (and two devs) preferring Solution 3, that's what we'll go with.
